pgoapi/pgoapi.py

```diff
@@ -541,11 +541,22 @@ def cleanup_pokemon(self, inventory_items=None):
         caught_pokemon = self.get_caught_pokemons(inventory_items)
         for pokemons in caught_pokemon.values():
             if len(pokemons) > self.MIN_SIMILAR_POKEMON:
-                # highest lvl pokemon first
-                sorted_pokemons = sorted(pokemons, key=self.pokemon_lvl, reverse=True)
-                for pokemon in sorted_pokemons[self.MIN_SIMILAR_POKEMON:]:
-                    if self.is_pokemon_eligible_for_transfer(pokemon, sorted_pokemons[0]):
-                        self.do_release_pokemon(pokemon)
+                if self.RELEASE_DUPLICATES:
+                    sorted_pokemons = sorted(pokemons, key=self.pokemon_lvl, reverse=True)
+
+                    for pokemon in sorted_pokemons[self.MIN_SIMILAR_POKEMON:]:
+                        if self.is_pokemon_eligible_for_transfer(pokemon, sorted_pokemons[0]):
+                            self.do_release_pokemon(pokemon)
+                else:
+                    remaining = len(pokemons)
```

Comment:
also here, if we really intent to keep the pokemon under simple CP IV rules, sort them by (cp,iv) and take away the first MIN_SIMILAR_POKEMON and then see if the others are eligible for transfer. 
If you really wanted to distinguish b/w  RELEASE_DUPLICATES a better strategy would be 
if Release_dups:
    sorted_mons  = sort by lvl
else
    sorted_mons = sort by cp,iv
for each in sorted_mons [MIN_SIMILAR_POKEMON:]:
    is_pok_eleible(pok, sorted_mons[0]):
        transfer.